web_sdk/make_combined_sdk.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:dev_compiler/src/analyzer/command.dart' as command; // ignore: uri_does_not_exist
+
+// Script for creating flutter analyzer summmary and precompiled SDK.
+Future<void> main() async {
+  // create a temporary dart-sdk directory.
+  final Directory tempDartSdk = Directory(path.join('temp_dart_sdk', 'lib'))
+    ..createSync(recursive: true);
+  final Directory patchedDartSdk = Directory(path.join(
+      'gen', 'third_party', 'dart', 'utils', 'dartdevc', 'patched_sdk', 'lib'));
+  final Directory flutterWebUi = Directory(path.join(
+    'flutter_web_sdk',
+    'lib',
+    'ui',
+  ));
+
+  // Copy this patched dart sdk into the temporary directory.
+  for (FileSystemEntity entity in patchedDartSdk.listSync(recursive: true)) {
+    if (entity is File) {
+      final String targetPath = path.join(tempDartSdk.path,
+          path.relative(entity.path, from: patchedDartSdk.path));
+      File(targetPath).createSync(recursive: true);
+      entity.copySync(path.join(tempDartSdk.path,
+          path.relative(entity.path, from: patchedDartSdk.path)));
+    }
+  }
+  // Copy the dart:ui sources into the temporary directory.
+  for (FileSystemEntity entity in flutterWebUi.listSync(recursive: true)) {
+    if (entity is File) {
+      final String targetPath = path.join(tempDartSdk.path, 'ui',
+          path.relative(entity.path, from: flutterWebUi.path));
+      File(targetPath).createSync(recursive: true);
+      entity.copySync(targetPath);
+    }
+  }
+  // Copy the libraries.dart file into the temporary directory.
+  final File libraries = File(path.join('..', '..', 'flutter', 'web_sdk', 'libraries.dart'));
+  libraries.copySync(path.join(tempDartSdk.path, '_internal', 'libraries.dart'));
+  libraries.copySync(path.join(tempDartSdk.path, '_internal', 'sdk_library_metadata', 'lib', 'libraries.dart'));
+
+  // Prevent regular compilation from leaking into flutter
+  final File ddcSummary =
+      File(path.join(tempDartSdk.path, '_internal', 'ddc_sdk.sum'));
+  final File jsSdk =
+      File(path.join(tempDartSdk.parent.path, 'js', 'amd', 'dart_sdk.js'));
+  final File jsSdkMap =
+      File(path.join(tempDartSdk.parent.path, 'js', 'amd', 'dart_sdk.js.map'));
+  if (ddcSummary.existsSync()) {
```

Comment:
Why do we need to check for their existence? Are these things not always there? Maybe instead _assert_ that they exist?

Reply:
I might be paranoid, but I was noticing the analyzer not overwriting the existing files...